### PR TITLE
Documentation for multiple updates for a buffer

### PR DESCRIPTION
**Pull request by jaebaek**
Related to #197

### CODE REVIEW

File: docs/multiple_updates_for_same_buffer.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                        ... ( more than 4000 numbers) ... 4096.0
+compute 1 1 1
+probe ssbo float 0 ~= 2.0
+ssbo 0 subdata float 4000 1.0 2.0 3.0 \
+                          ... ( more than 4000 numbers) ... 4096.0
+compute 1 1 1
+probe ssbo float 4000 ~= 1.0
+```
+
+This script has the same behavior as the following script does
```
Comment (dj2) [inline docs/multiple_updates_for_same_buffer.md]: That's not quite correct. It uses all of them up to the probe. The probe commands submit the command buffer which collapses all the ssbo buffers. So, the first 3 ssbo commands will be used with the first 3 compute calls. The 4th ssbo command will be used with the 4th compute command.

File: docs/multiple_updates_for_same_buffer.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+```
+ssbo 0 subdata float 0 1.0
+ssbo 0 subdata float 0 2.0
+ssbo 0 subdata float 36 1.0 2.0 3.0 \
+                        ... ( more than 4000 numbers) ... 4096.0
+ssbo 0 subdata float 4000 1.0 2.0 3.0 \
```
Comment (dj2) [inline docs/multiple_updates_for_same_buffer.md]: This should be after the first probe

File: docs/multiple_updates_for_same_buffer.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+* the second one sees `2.0` if the first compute does not change the
+  SSBO
+* the third one sees
+  `2.0 0 0 ... 0 1.0 2.0 3.0 ... ( more than 4000 numbers) ... 4096.0`
+  if the first and second computes does not change the SSBO
+
+If each `compute` writes something to SSBO, it also remains in the SSBO.
+
+
+### How to solve mismatch between those two scripts
```
Comment (dj2) [inline docs/multiple_updates_for_same_buffer.md]: I don't think we should do anything. Just run it, with a note in the README that the behaviour is different. I don't think we want to be adjusting the script on the fly.

File: docs/multiple_updates_for_same_buffer.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+compute 1 1 1
+compute 1 1 1
+probe ssbo float 0 ~= 2.0
+compute 1 1 1
+probe ssbo float 4000 ~= 1.0
+```
+
+### Multiple updates for a SSBO in Amber script
+
+Amber script will have different semantics for those multiple
```
Comment (dneto0) [inline docs/multiple_updates_for_same_buffer.md]: If Amber is running a vkscript then it should behave the same as VkRunner.

File: docs/multiple_updates_for_same_buffer.md (hunk 3 above)
Comment (jaebaek) [reply docs/multiple_updates_for_same_buffer.md]: As @dneto0 mentioned, Amber must behave the same as VkRunner does if it is VkRunner script. However, Amber script will have different semantics to make users more clear. To overcome this mismatch, I recommend that we have slight different executors for Amber script and VkRunner script while we do not change Vulkan engine.

File: docs/multiple_updates_for_same_buffer.md (hunk 4 above)
Comment (dj2) [reply docs/multiple_updates_for_same_buffer.md]: I don't think we want follow what VkRunner does here as the script is confusing. I think we should deviate from VkRunner in this case. We just have to document that we have deviated.

**Comment by dj2** [comment]
I'm going to close this for now. Please re-open if it's still relevant.

**Pull request closed without merging**